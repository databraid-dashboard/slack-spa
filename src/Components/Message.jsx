// @flow

import React from 'react';
import { Image, List } from 'semantic-ui-react';
import type { MessageType } from '../FlowTypes/';

const moment = require('moment');

/*
NOTE: Properties available for each message:
- messageId* (integer): Primary key from database; Property
  of messages retrieved from the database, not on messages
  emitted via Socket.io
- avatarImage (string): Image size 24
- name (string): "Real name", e.g. 'Kurtis Houser'
- userName (string): "User name", e.g. 'thekurtishouser' or other
- text (string): The message
- timestamp (string): Formatted timestamp, e.g. '2017-08-01T22:20:43.643Z'
- rawTimestamp (string): Raw timestamp e.g. '1501626043.643661'
- channelName (string): Name of the the channel, e.g. 'dev'
- statusEmoji (string): As labeled internally by slack, e.g. ':slack:'
*/

export default function Message({ avatarImage, name, text, timestamp }: MessageType) {
  return (
    <List.Item className="listItem">
      <Image avatar src={avatarImage} />
      <List.Content>
        <List.Header>
          <span className="message-author">{name}</span>
          <span className="message-timestamp">
            {`\t${moment(timestamp).format('LLLL')}`}
          </span>
          <List.Description className="message-text">{text}</List.Description>
        </List.Header>
      </List.Content>
    </List.Item>
  );
}
