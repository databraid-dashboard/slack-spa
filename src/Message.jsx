// @flow

import React from 'react';
import { Image, List } from 'semantic-ui-react';

import type { MessageType } from './store';

export default function Message(
  { avatarImage, name, text, timestamp }: MessageType,
) {
  return (
    <List.Item className="listItem">
      <Image avatar src={avatarImage} />
      <List.Content>
        <List.Header>
          {`${name}, ${timestamp}`}
          <List.Description>
            {text}
          </List.Description>
        </List.Header>
      </List.Content>
    </List.Item>
  );
}
