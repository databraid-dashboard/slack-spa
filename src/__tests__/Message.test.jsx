import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Message from '../Components/Message';

describe('Renders message with default props', () => {
  it('should render a message when given props', () => {
    const message = shallow(
      <Message
        avatarImage="img.png"
        name="John Appleseed"
        text="Hello there"
        timestamp="3:50 PM, 08-02-2017"
      />,
    );
    expect(toJson(message)).toMatchSnapshot();
  });
});
