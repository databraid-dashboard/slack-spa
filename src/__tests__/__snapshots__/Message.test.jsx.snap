// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders message with default props should render a message when given props 1`] = `
<ListItem
  className="listItem"
>
  <Image
    as="img"
    avatar={true}
    ui={true}
  />
  <ListContent>
    <ListHeader>
      John Appleseed
       
      3:50 PM, 08-02-2017
      <ListDescription>
        Hello there
      </ListDescription>
    </ListHeader>
  </ListContent>
</ListItem>
`;
